File: android/app/src/main/java/pl/kinowo/ui/dev/DisplayInfo.kt
package pl.kinowo.ui.dev

import kotlin.math.roundToInt

/**
 * Read-only display-geometry readout for the dev tuning screen. It leads with
 * the *available* logical size that actually drives layout (dp) — more useful
 * than the raw panel resolution for dialling in the two-per-row showtime fit —
 * then the display density and the physical pixel resolution it works out to.
 *
 * Pure → JVM-unit-tested without an emulator; the Compose screen feeds it
 * `LocalConfiguration` dp + `LocalDensity` density.
 */
object DisplayInfo {
    /** e.g. `viewport 411×891 dp · 2.625× → 1079×2339 px` */
    fun tuningReadout(widthDp: Int, heightDp: Int, density: Float): String {
        val px = (widthDp * density).roundToInt()
        val py = (heightDp * density).roundToInt()
        return "viewport ${widthDp}×${heightDp} dp · ${densityText(density)}× → ${px}×${py} px"
    }

    /** `3.0` → `3`, `2.625` → `2.625` — drop a trailing `.0`, keep real fractions. */
    private fun densityText(density: Float): String =
        if (density == density.toInt().toFloat()) density.toInt().toString() else density.toString()
}

File: ios/Kinowo/DisplayInfo.swift
import Foundation

/// Read-only display-geometry readout for the dev tuning screen. It leads with
/// the *available* logical size that actually drives layout (points on iOS) —
/// more useful than the raw panel resolution for dialling in the two-per-row
/// showtime fit — then the screen scale and the physical pixel resolution it
/// works out to.
///
/// Pure + Foundation-only so it lives in `KinowoCore` and is unit-tested on
/// Linux; the SwiftUI screen feeds it `GeometryReader` points + `UIScreen`
/// scale.
enum DisplayInfo {
    /// e.g. `viewport 390×844 pt · @3× → 1170×2532 px`
    static func tuningReadout(pointWidth: Double, pointHeight: Double, scale: Double) -> String {
        let width = Int(pointWidth.rounded())
        let height = Int(pointHeight.rounded())
        let pixelWidth = Int((pointWidth * scale).rounded())
        let pixelHeight = Int((pointHeight * scale).rounded())
        return "viewport \(width)×\(height) pt · @\(scaleText(scale))× → \(pixelWidth)×\(pixelHeight) px"
    }

    /// `2.0` → `2`, `2.625` → `2.625` — drop a trailing `.0`, keep real fractions.
    private static func scaleText(_ scale: Double) -> String {
        String(format: "%g", scale)
    }
}
